TACOS-rest/src/main/java/cz/fi/muni/TACOS/rest/controllers/ProductCategoryController.java

```diff
@@ -0,0 +1,179 @@
+package cz.fi.muni.TACOS.rest.controllers;
+
+import cz.fi.muni.TACOS.dto.ProductCategoryCreateDTO;
+import cz.fi.muni.TACOS.dto.ProductCategoryDTO;
+import cz.fi.muni.TACOS.exceptions.InvalidRelationEntityIdException;
+import cz.fi.muni.TACOS.facade.ProductCategoryFacade;
+import cz.fi.muni.TACOS.facade.ProductFacade;
+import cz.fi.muni.TACOS.rest.exceptions.InvalidParameterException;
+import cz.fi.muni.TACOS.rest.exceptions.ResourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+/**
+ * @author Peter Balcirak <peter.balcirak@gmail.com>
+ */
+public class ProductCategoryController {
+
+	private static final Logger log = LoggerFactory.getLogger(AttributeCategoryController.class);
+
+	@Inject
+	private ProductCategoryFacade productCategoryFacade;
+
+	@Inject
+	private ProductFacade productFacade;
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	public List<ProductCategoryDTO> getAllProductCategories() {
+		log.debug("rest getAllProductCategories()");
+
+		return productCategoryFacade.getAll();
+	}
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("/{id}")
+	public ProductCategoryDTO findProductCategoryById(@PathParam("id") Long id) {
+		log.debug("rest findProductCategoryById({})", id);
+
+		ProductCategoryDTO productCategoryDTO = productCategoryFacade.findById(id);
+
+		if (productCategoryDTO == null) {
+			throw new InvalidParameterException("Product category for given id does not exist. id: " + id);
+		}
+
+		return productCategoryDTO;
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Long createProductCategory(ProductCategoryCreateDTO specification) {
+		log.debug("rest createProductCategory({})", specification);
+
+		validateProductCategory(specification);
+
+		try {
+			return productCategoryFacade.create(specification);
+		} catch (InvalidRelationEntityIdException e) {
+			throw new InvalidParameterException(e);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/{parentId}/createSubCategory")
+	public Long createSubCategory(ProductCategoryCreateDTO specification, @PathParam("parentId") Long parentId) {
```

Comment:
This method is redundant, if u want to create subcategory u just create a categroy with specified parent category.